item.lua

```diff
@@ -0,0 +1,300 @@
+-----------------------------------------------------------
+-- helper functions that should probably be moved elsewhere
+-----------------------------------------------------------
+
+-- check whether an item is inside a burrow
+local function containsItem(burrow,item)
+    local res = false
+    local x,y,z = dfhack.items.getPosition(item)
+    if x then
+        res = dfhack.burrows.isAssignedTile(burrow, xyz2pos(x,y,z))
+    end
+    return res
+end
+
+-- check whether the item is reachable by one of the citizens
+local function isReachable(item)
+    local reachable = false
+    local x, y, z = dfhack.items.getPosition(item)
+    if x then -- item has a valid position
+        local citizens = dfhack.units.getCitizens(true)
+        for _, unit in pairs(citizens) do
+            if dfhack.maps.canWalkBetween(unit.pos, xyz2pos(x, y, z)) then
+                reachable = true
+                break
+            end
+        end
+    end
+    return reachable
+end
+
+-- fast reachability test for items that requires precomputed walkability groups for
+-- all citizens. Returns false for items w/o valid position (e.g., items in inventories).
+--- @param item item
+--- @param wgroups table<integer,boolean>
+--- @return boolean
+local function fastReachable(item,wgroups)
+    local x, y, z = dfhack.items.getPosition(item)
+    if x then -- item has a valid position
+        local igroup = dfhack.maps.getWalkableGroup(xyz2pos(x, y, z))
+        if wgroups[igroup] then
+            return true
+        else
+            return false
+        end
```

Comment:
if returning something truthy is enough, consider `return wgroups[igroup]`. If it needs to be a boolean, consider `return not not wgroups[igroup]`